src/virtualenv/activation/bash/__init__.py

```diff
@@ -1,13 +1,35 @@
 from __future__ import absolute_import, unicode_literals
 
+from virtualenv.info import IS_WIN
 from virtualenv.util.path import Path
+from virtualenv.util.six import ensure_text
 
 from ..via_template import ViaTemplateActivator
 
 
 class BashActivator(ViaTemplateActivator):
+    def generate(self, creator):
+        generated = super(BashActivator, self).generate(creator)
+        if IS_WIN:
```

Comment:
Why do this conversion and not just always use linux line endings? 🤔

Reply:
To keep the change localized to `BashActivator`, else we'd have to make the change in `ViaTemplateActivator` which will affect many activators.

Reply:
Seems the line normalization happens because we use read_text/write_text. In case of these activator templates I don't think we want any of newline normalization, so a better solution is to just switch to reading binary data for the templates, and encode it as UTF-8 ourselves.